src/pages/Summary/index.js

```diff
@@ -1,11 +1,139 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import { selectToken } from '../../store/user/selectors'
+import {
+	selectUserRecords,
+	selectCountOfUserRecords,
+} from '../../store/record/selectors'
+import { fetchAllUserRecords } from '../../store/record/actions'
+import Loading from '../../components/Loading'
+
+import './SummaryPage.css'
 
 const Summary = () => {
-  return (
-    <div>
-      Summary
-    </div>
-  )
+	const dispatch = useDispatch()
+	const history = useHistory()
+	const token = useSelector(selectToken)
+	const records = useSelector(selectUserRecords)
+	const count = useSelector(selectCountOfUserRecords)
+
+	useEffect(() => {
+		if (records.length === 0) {
+			dispatch(fetchAllUserRecords())
+		}
+	}, [])
+
+	if (!records || records.length === 0) {
+		return <Loading />
+	}
+
+	if (!token) {
+		history.push('/')
+	}
+	const sortedPrice = [...records].sort((a, b) => b.lowestPrice - a.lowestPrice)
+
+	const worthOne = sortedPrice.length >= 1 ? sortedPrice[0] : null
+	const worthTwo = sortedPrice.length >= 2 ? sortedPrice[1] : null
+	const worthThree = sortedPrice.length >= 3 ? sortedPrice[2] : null
+
+	const totalPrice = records
+		.map((record) => record.lowestPrice)
+		.reduce((a, b) => a + b, 0)
+
+	const oldestYear = Math.min.apply(
+		Math,
+		records.map((record) => record.year)
+	)
+
+	const genres = records.flatMap((record) => record.genre.split(' | '))
+
+	const aCount = new Map(
+		[...new Set(genres)].map((x) => [x, genres.filter((y) => y === x).length])
+	)
+
+	const countedGenres = genres
+		.map((genre) => {
+			return {
+				genre: genre,
+				count: aCount.get(genre),
+			}
+		})
```

Comment:
```
genres.map(genre => ({ genre, count: aCount.get(genre) }))
```

Reply:
if your returning straight away no need for curly brackets. you use straight away normal parenthesis and what you want to return